mountaineer/__tests__/test_annotation_helpers.py

```diff
@@ -1,14 +1,27 @@
+import sys
 from dataclasses import dataclass
 
 from pydantic import BaseModel
 
 from mountaineer.annotation_helpers import yield_all_subtypes
 
+if sys.version_info >= (3, 11):
+    from enum import StrEnum
+else:
+    from backports.strenum import StrEnum
```

Comment:
I _think_ we should be able to get the necessary behavior of StrEnum just by inheriting from both superclasses `MyEnum(str, Enum)`. There's a bit of additional logic that the actual StrEnum class introduces for overriding magic functions, of course, but we're mostly using them for transparent python->openapi generation.

Instead of another dependency I'd suggest a `mountaineer.compat` implementation of StrEnum where required / created based on an import conditional for versions where StrEnum isn't available natively.